en/reference/component-reference.html

```diff
@@ -3,182 +3,103 @@
-  A component may be injected into other components,
-  including all the different request handlers, document processors and so on mentioned above,
-  and will get a destructor like method invoked when no other component is using it any more.
+  Whenever a component or a resource your component depends on is changed by a redeployment,
+  your component is reconstructed. Once all changed components are reconstructed, new requests
+  are atomically switched to use the new set and the old ones are destructed.
 </p>
 
+<p>
+  If you have multiple constructors in your component, annotate the one to use for injection by
+  <code><a href="https://google.github.io/guice/api-docs/latest/javadoc/com/google/inject/Inject.html">@Inject</a></code>.
+</p>
+
+<p>
+  Identifiable components must implement com.yahoo.component.Component, and components that need to
+  destruct resources at removal must subclass com.yahoo.component.AbstractComponent and implement deconstruct().
+</p>
+
+
+<h2 id="component-types">Component Types</h2>
+
+<p>
+  Vespa defined various component types (superclasses) for common tasks:
+</p>
 
 <table class="table">
   <tr id="request-handlers">
     <th>Request handlers</th>
     <td>
       <p>
-        Request handlers are the easiest way to implement arbitrary REST services.
-        The incoming request is a set of key-value pairs and protocol specific information.
-        A request handler my fire off a request to e.g. the search handler,
-        but what it will get back is opaque.
-        If context sensitive result transformations are necessary,
-        the request handler must be paired with one or more <a href="#searchers">searchers</a>,
-        and if the output from the search needs to be customized,
-        a <a href="#renderers">renderer</a> for the search result is the correct way of doing it
-        (as opposed to trying to transform the search response in the request handler).
-      </p><p>
-        A request handler accepts a request, and returns a response.
-        All applications will use a request handler, either by implementing their own,
-        or indirectly by using the <a href="#processor-instances">processing framework</a>,
-        as the processing handler is a request handler as well.
-        A client provider is-a request handler.
-      </p><p>
-        Requests are modelled as key-value stores, as in many other systems,
-        while responses are meta information (like HTTP status)
-        and a binary stream of content (the <em>ContentChannel</em> API).
-      </p><p>
-        A content channel is the basic I/O building block of JDisc.
-        It is conceptually similar to a stream,
-        but the basic unit is the ByteBuffer instead of the native byte.
-        A content channel will take ownership of a ByteBuffer written to it.
+        Request handlers lets applications implement arbitrary HTTP API's.
       </p>
-    </td>
-  </tr>
-  <tr id="response-handlers">
```

Comment:
Not sure if Response handler is a valid concept anymore / should be documented - ref https://github.com/vespa-engine/documentation/blob/master/en/reference/component-reference.html#L203 (broken link) and the illustration that still uses this name. Is it correct to just use the term Request handler everywhere?

Reply:
They are a valid concept for but not a component type, which made this confusing since it purports to be a list of component types. However, later in this I gave up trying to make this an actual list of components.

They are only relevant for people who want to write their own async handlers in jDisc, which is very far from what should be covered in a "component reference". The real problem here is that this documents is, in addition to containing info on components, is trying to teach about the working of jdisc core. 

I ran out of steam before cleaning that up completely, and I think. it's ok to leave it like it is, but if you want to bring the response handler doc back I guess that's fine too.

Reply:
Thanks! I will just remove the broken link for now - this doc is much better than it used to be :-)